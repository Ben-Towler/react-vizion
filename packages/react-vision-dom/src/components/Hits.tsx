import React from 'react';
import classnames from 'classnames';
import { createClassNames } from '../core/utils';

export type SearchResult = {
  id: string;
};

type HitProps = {
  results: SearchResult[];
  className?: string;
  style?: React.CSSProperties;
  noResultsFound?: React.ReactNode;
  hit?: (result: SearchResult) => React.ReactNode;
  onClick?: (event?: React.MouseEvent) => void;
};

type DefaultHitProps = {
  searchResult: SearchResult;
};

const DefaultHit: React.FunctionComponent<DefaultHitProps> = ({
  searchResult,
}) => {
  return (
    <div className="default-hit-card">
      <div className="default-hit-container">
        <h4 className="default-hit-title">{searchResult.id}</h4>
        <div className="card-body">{JSON.stringify(searchResult, null, 2)}</div>
      </div>
    </div>
  );
};

const cx = createClassNames('Hits');

const Hits: React.FunctionComponent<HitProps> = ({
  results,
  className,
  style,
  hit,
  noResultsFound,
  onClick,
}) => {
  return (
    <div className={classnames(cx('hits'), className)} style={style}>
      <div className={cx('hits-list')}>
        {Array.isArray(results) && results.length > 0
          ? results.map(searchResult => (
              <div key={searchResult.id} onClick={onClick}>
                {hit ? (
                  hit(searchResult)
                ) : (
                  <>
                    <DefaultHit searchResult={searchResult} />
                  </>
                )}
              </div>
            ))
          : noResultsFound || (
              //Temporarily until we have the styles decided
              <div className={cx('empty')}>No results were found</div>
            )}
      </div>
    </div>
  );
};

export default Hits;
