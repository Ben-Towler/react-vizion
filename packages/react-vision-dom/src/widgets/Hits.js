import { connectHits } from '@clinia/react-vision-core';
import Hits from '../components/Hits';

/**
 * Displays a list of hits.
 *
 * To configure the number of hits being shown, use the [HitsPerPage widget](widgets/HitsPerPage.html),
 * [connectHitsPerPage connector](connectors/connectHitsPerPage.html) or the [Configure widget](widgets/Configure.html).
 *
 * @name Hits
 * @kind widget
 * @propType {Component} [hitComponent] - Component used for rendering each hit from
 *   the results. If it is not provided the rendering defaults to displaying the
 *   hit in its JSON form. The component will be called with a `hit` prop.
 * @themeKey cvi-Hits - the root div of the widget
 * @themeKey cvi-Hits-list - the list of results
 * @themeKey cvi-Hits-item - the hit list item
 * @example
 * import React from 'react';
 * import clinia from 'clinia/lite';
 * import { Vision, Hits } from '@clinia/react-vision-dom';
 *
 * const searchClient = clinia(
 *   'latency',
 *   '6be0576ff61c053d5f9a3225e2a90f76'
 * );
 * const App = () => (
 *   <Vision
 *     searchClient={searchClient}
 *     indexName="instant_search"
 *   >
 *     <Hits />
 *   </Vision>
 * );
 */

export default connectHits(Hits);
