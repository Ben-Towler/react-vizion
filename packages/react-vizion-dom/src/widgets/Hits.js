import { connectHits } from '@clinia/react-vizion-core';
import Hits from '../components/Hits';

/**
 * Displays a list of hits.
 *
 * To configure the number of hits being shown, use the [PerPage widget](widgets/PerPage.html),
 * [connectPerPage connector](connectors/connectPerPage.html) or the [Configure widget](widgets/Configure.html).
 *
 * @name Hits
 * @kind widget
 * @propType {Component} [hitComponent] - Component used for rendering each hit from
 *   the results. If it is not provided the rendering defaults to displaying the
 *   hit in its JSON form. The component will be called with a `hit` prop.
 * @themeKey cvi-hits - the root div of the widget
 * @themeKey cvi-hits-list - the list of results
 * @themeKey cvi-hits-item - the hit list item
 * @example
 * import React from 'react';
 * import clinia from 'clinia/lite';
 * import { Vizion, Hits } from '@clinia/react-vizion-dom';
 *
 * const searchClient = clinia(
 *   'latency',
 *   '6be0576ff61c053d5f9a3225e2a90f76'
 * );
 * const App = () => (
 *   <Vizion
 *     searchClient={searchClient}
 *     indexName="instant_search"
 *   >
 *     <Hits />
 *   </Vizion>
 * );
 */

export default connectHits(Hits);
