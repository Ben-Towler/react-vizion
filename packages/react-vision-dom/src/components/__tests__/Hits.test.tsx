import React from 'react';
import renderer from 'react-test-renderer';
import Hits from '../Hits';

describe('Hits', () => {
  type Props = { hit: any };

  const Hit = ({ hit }: Props) => <div id={hit.id} />;

  it('accepts a hitComponent prop', () => {
    const hits = [{ id: 0 }, { id: 1 }, { id: 2 }];

    const tree = renderer.create(<Hits hitComponent={Hit} hits={hits} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('accepts a custom className', () => {
    const hits = [{ id: 0 }, { id: 1 }, { id: 2 }];

    const tree = renderer.create(
      <Hits className="MyCustomHits" hitComponent={Hit} hits={hits} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
