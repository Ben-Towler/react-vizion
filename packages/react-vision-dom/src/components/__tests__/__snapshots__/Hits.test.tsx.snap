// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hits accepts a custom className 1`] = `
<div
  className="cvi-Hits MyCustomHits"
>
  <ul
    className="cvi-Hits-list"
  >
    <li
      className="cvi-Hits-item"
    >
      <div
        id={0}
      />
    </li>
    <li
      className="cvi-Hits-item"
    >
      <div
        id={1}
      />
    </li>
    <li
      className="cvi-Hits-item"
    >
      <div
        id={2}
      />
    </li>
  </ul>
</div>
`;

exports[`Hits accepts a hitComponent prop 1`] = `
<div
  className="cvi-Hits"
>
  <ul
    className="cvi-Hits-list"
  >
    <li
      className="cvi-Hits-item"
    >
      <div
        id={0}
      />
    </li>
    <li
      className="cvi-Hits-item"
    >
      <div
        id={1}
      />
    </li>
    <li
      className="cvi-Hits-item"
    >
      <div
        id={2}
      />
    </li>
  </ul>
</div>
`;
